protobuf-codegen-pure/src/parser.rs

```diff
@@ -974,12 +974,12 @@ impl<'a> Parser<'a> {
             let name = self.tokenizer.next_ident()?;
             self.tokenizer.next_symbol_expect_eq('(')?;
             let client_streaming = self.tokenizer.next_ident_if_eq("stream")?;
-            let input_type = self.tokenizer.next_ident()?;
+            let input_type = self.next_message_or_enum_type()?;
```

Comment:
I think semantically these are only allowed to be message types (not enums), but syntactically it's all the same so this doesn't matter?